pageserver/src/tenant/remote_timeline_client/index.rs

```diff
@@ -55,6 +60,10 @@ pub struct IndexPart {
     #[serde(default)]
     version: usize,
 
+    #[serde(default)]
+    #[serde(skip_serializing_if = "is_false")]
+    pub is_deleted: bool,
```

Comment:
On more idea:

In generic app development an `is_deleted` flag lifecycle usually ends up changed to `deleted_at: Option<Instant>` (or SystemTime or whatever) rather quickly as debugging starts.

Should we jump the shark/gun/"for future" because we could theoretically use the datascrubber to notice when timelines are deleted in s3 in interesting delays (or even future) compared to the control plane database?

This of course would be automatic in a more event based storage :)

Reply:
Good idea, done in https://github.com/neondatabase/neon/pull/3919/commits/5b6853b50d6765c190d4212030b8ab5cd32f26e5